group-coordinator/src/main/java/org/apache/kafka/coordinator/group/runtime/CoordinatorRuntime.java

```diff
@@ -1046,12 +1118,18 @@ public void run() {
                             result
                         );
 
-                        long offset = partitionWriter.appendEndTransactionMarker(
+                        long offset = partitionWriter.append(
                             tp,
-                            producerId,
-                            producerEpoch,
-                            coordinatorEpoch,
-                            result
+                            VerificationGuard.SENTINEL,
+                            MemoryRecords.withEndTransactionMarker(
+                                time.milliseconds(),
```

Comment:
It seems we didn't specify this time value before. Was that a bug? I guess it also just gets the system time in the method.

Reply:
`withEndTransactionMarker` takes the current time if we don't specify it. The reason why I set it explicitly here is to ensure that the mock time is used in tests.